### PR TITLE
Added documentation for new System.Net APIs released in 3.0

### PR DESCRIPTION
**Pull request by jozkee**
List of APIs documented:

- CookieCollection:

1. M:System.Net.CookieCollection.Clear
2. M:System.Net.CookieCollection.Contains(System.Net.Cookie)
3. M:System.Net.CookieCollection.Remove(System.Net.Cookie)
4. M:System.Net.CookieCollection.System#Collections#Generic#IEnumerable<System#Net#Cookie>#GetEnumerator

- FileWebRequest:

1. M:System.Net.FileWebRequest.GetRequestStreamAsync
2. M:System.Net.FileWebRequest.GetResponseAsync

- IPEndPoint:

1. M:System.Net.IPEndPoint.Parse(System.ReadOnlySpan{System.Char})
2. M:System.Net.IPEndPoint.Parse(System.String)
3. M:System.Net.IPEndPoint.TryParse(System.String,System.Net.IPEndPoint@)
4. M:System.Net.IPEndPoint.TryParse(System.ReadOnlySpan{System.Char},System.Net.IPEndPoint@)

### CODE REVIEW

**Comment by Jozkee** [comment]
@davidsh @wfurt @krwq @scalablecory  since you are the System.Net owners, could you please take a look?

**Comment by Jozkee** [comment]
@mairaw @rpetrusha

File: xml/System.Net/CookieCollection.xml [hunk 1]
```diff
@@ -291,10 +291,18 @@
         <Parameter Name="cookie" Type="System.Net.Cookie" Index="0" FrameworkAlternate="netcore-3.0" />
       </Parameters>
       <Docs>
-        <param name="cookie">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="cookie">The <see cref="T:System.Net.Cookie" /> to locate in the <see cref="T:System.Net.CookieCollection" />.</param>
+        <summary>Determines whether the <paramref name="cookie" /> is in the <see cref="T:System.Net.CookieCollection" />.</summary>
+        <returns><see langword="true" /> if <paramref name="cookie" /> is found in the <see cref="T:System.Net.CookieCollection" />; otherwise, is <see langword="false" />.</returns>
```
Comment (scalablecory) [inline xml/System.Net/CookieCollection.xml]: > otherwise, is false

you've got stray "is"

File: xml/System.Net/CookieCollection.xml [hunk 2]
```diff
@@ -778,10 +786,20 @@
         <Parameter Name="cookie" Type="System.Net.Cookie" Index="0" FrameworkAlternate="netcore-3.0" />
       </Parameters>
       <Docs>
-        <param name="cookie">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="cookie">The <see cref="T:System.Net.Cookie" /> to remove form the <see cref="T:System.Net.CookieCollection" />.</param>
+        <summary>Determines whether the <paramref name="cookie" /> is in the <see cref="T:System.Net.CookieCollection" />.</summary>
+        <returns><see langword="true" /> if item was successfully removed from the <see cref="T:System.Net.CookieCollection" />; otherwise, <see langword="false" />. This method also returns <see langword="false" /> if item is not found in the original collection.</returns>
```
Comment (scalablecory) [inline xml/System.Net/CookieCollection.xml]: > if item is not found in the original collection

Was this copied from `ICollection`? There is no "item" parameter here.

File: xml/System.Net/FileWebRequest.xml [hunk 3]
```diff
@@ -717,9 +717,29 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Returns a <see cref="T:System.IO.Stream" /> object for writing data to the file system resource as an asynchronous operation.</summary>
+        <returns>The Task object representing the asynchronous operation.</returns>
```
Comment (scalablecory) [inline xml/System.Net/FileWebRequest.xml]: "Task" -> `<see cref="...Task"/>`.

File: xml/System.Net/IPEndPoint.xml [hunk 4]
```diff
@@ -568,10 +568,12 @@
         <Parameter Name="s" Type="System.ReadOnlySpan&lt;System.Char&gt;" Index="0" FrameworkAlternate="netcore-3.0" />
       </Parameters>
       <Docs>
-        <param name="s">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="s">A <see cref="T:System.RedOnlySpan`1" /> that contains an IP endpoint in dotted-quad notation or network byte order for IPv4 and in colon-hexadecimal notation for IPv6.</param>
+        <summary>Converts an IP network endpoint (address and port) represented as a <see cref="T:System.RedOnlySpan`1" /> to an IPEndPoint instance.</summary>
```
Comment (scalablecory) [inline xml/System.Net/IPEndPoint.xml]: RedOnlySpan -> ReadOnlySpan

File: xml/System.Net/IPEndPoint.xml [hunk 5]
```diff
@@ -568,10 +568,12 @@
...
+        <returns>An <see cref="T:System.Net.IPEndPoint" /> instance.</returns>
         <remarks>To be added.</remarks>
```
Comment (scalablecory) [inline xml/System.Net/IPEndPoint.xml]: Remarks still todo?

File: xml/System.Net/FileWebRequest.xml [hunk 6]
```diff
@@ -799,9 +819,22 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Returns a response to a file system request as an asynchronous operation.</summary>
+        <returns>The Task object representing the asynchronous operation.</returns>
```
Comment (scalablecory) [inline xml/System.Net/FileWebRequest.xml]: "Task" -> `<see cref="...Task"/>`.

File: xml/System.Net/FileWebRequest.xml [hunk 7]
```diff
@@ -717,9 +717,29 @@
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Returns a <see cref="T:System.IO.Stream" /> object for writing data to the file system resource as an asynchronous operation.</summary>
+        <returns>The Task object representing the asynchronous operation.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+          ## Remarks  
+          > [!NOTE]
+          >  In the case of asynchronous requests, it is the responsibility of the client application to implement its own time-out mechanism.
```
Comment (scalablecory) [inline xml/System.Net/FileWebRequest.xml]: Should this be a link to something describing how to implement time-outs?

File: xml/System.Net/IPEndPoint.xml (hunk 5 above)
Comment (Jozkee) [reply xml/System.Net/IPEndPoint.xml]: Since overloads are generated in the same html, for this case and `TryParse` I think it would be redundant to have practically the same remarks twice.

File: xml/System.Net/FileWebRequest.xml (hunk 7 above)
Comment (Jozkee) [reply xml/System.Net/FileWebRequest.xml]: there is an example for [`HttpWebRequest.BeginGetResponse`](https://docs.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest.begingetresponse?view=netcore-3.0) I could adapt it to `FileWebRequest.GetResponseAsync` in case is really needed.

**Comment by mairaw** [comment]
You can see when your changes on the staging site: https://review.docs.microsoft.com/en-us/dotnet/api/system.net.ipendpoint?view=netcore-3.0&branch=pr-en-us-2663

File: xml/System.Net/IPEndPoint.xml [hunk 8]
```diff
@@ -602,10 +604,30 @@
...
+
+|Number of parts and example `s`|IPv4 address for IPEndPoint.Address|Port
+|-------------------------|--------------------------------|-------------------------|  
+|1 -- "1"|0.0.0.1|0|  
+|2 -- "20.2:80"|20.0.0.2|80|  
+|2 -- "20.65535:23"|20.0.255.255|23|  
+|3 -- "128.1.2:443"|128.1.0.2|443|  
+
+Note that this method accepts as valid an string value that can be parsed as an Int64, and then treats that Int64 as the long value of an IP address in network byte order, similar to the way that the IPAddress constructor does. This means that this method returns true if the Int64 is parsed successfully, even if it represents an address that's not a valid IP address. For example, if s is "1", this method returns true even though "1" (or 0.0.0.1) is not a valid IP address and you might expect this method to return false. Fixing this bug would break existing apps, so the current behavior will not be changed. Your code can avoid this behavior by ensuring that it only uses this method to parse IP addresses in dotted-decimal format.
```
Comment (mairaw) [inline xml/System.Net/IPEndPoint.xml]: This seems like an important note.

```suggestion
> [!IMPORTANT]
> Note that this method accepts as valid an string value that can be parsed as an <xref:System.Int64>, and then treats that <xref:System.Int64> as the long value of an IP address in network byte order, similar to the way that the <xref:System.Net.IPAddress> constructor does. This means that this method returns `true` when the <xref:System.Int64> is parsed successfully, even if it represents an address that's not a valid IP address. For example, if `s` is "1", this method returns `true` even though "1" (or 0.0.0.1) is not a valid IP address. Fixing this bug would break existing apps, so the current behavior will not be changed. Your code can avoid this behavior by ensuring that it only uses this method to parse IP addresses in dotted-decimal format.
```

File: xml/System.Net/IPEndPoint.xml [hunk 9]
```diff
@@ -793,10 +815,10 @@
...
       </Parameters>
       <Docs>
-        <param name="s">To be added.</param>
-        <param name="result">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="s">The IP endpoint to validate.</param>
+        <param name="result">The <see cref="T:System.Net.IPEndPoint" /> version of <paramref name="s" />.</param>
+        <summary>Determines whether <paramref name="s" /> is a valid IP endpoint.</summary>
+        <returns><see langword="true" /> if <paramref name="s" /> was able to be parsed as an IP endpoint; otherwise, <see langword="false" />.</returns>
```
Comment (mairaw) [inline xml/System.Net/IPEndPoint.xml]:
```suggestion
        <returns><see langword="true" /> if <paramref name="s" /> can be parsed as an IP endpoint; otherwise, <see langword="false" />.</returns>
```

File: xml/System.Net/IPEndPoint.xml [hunk 10]
```diff
@@ -829,11 +851,29 @@
...
       <Docs>
-        <param name="s">To be added.</param>
-        <param name="result">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="s">The IP endpoint to validate.</param>
+        <param name="result">The <see cref="T:System.Net.IPEndPoint" /> version of <paramref name="s" />.</param>
+        <summary>Determines whether <paramref name="s" /> is a valid IP endpoint.</summary>
+        <returns><see langword="true" /> if <paramref name="s" /> was able to be parsed as an IP endpoint; otherwise, <see langword="false" />.</returns>
```
Comment (mairaw) [inline xml/System.Net/IPEndPoint.xml]:
```suggestion
        <returns><see langword="true" /> if <paramref name="s" /> can be parsed as an IP endpoint; otherwise, <see langword="false" />.</returns>
```

File: xml/System.Net/IPEndPoint.xml [hunk 11]
```diff
@@ -829,11 +851,29 @@
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+The number of parts (each part is separated by a period) in the string s determines how the endpoint's IP address is constructed. A one part address is stored directly in the network address. A two part address, convenient for specifying a class A address, puts the leading part in the first byte and the trailing part in the right-most three bytes of the network address. A three part address, convenient for specifying a class B address, puts the first part in the first byte, the second part in the second byte, and the final part in the right-most two bytes of the network address. For example:
```
Comment (mairaw) [inline xml/System.Net/IPEndPoint.xml]: same comments as earlier for this set of remarks.

**Comment by davidsh** [comment]
All of these types and methods aren't really new.  What's new is that there are some overloads with ReadOnlySpan<T> instead of other buffer types. 

So, I would advise looking at that existing documentation to re-use most of the same descriptions of parameters, remarks, return values, etc. We only need small changes when new parameter types such as the ReadOnlySpan<T> are added.

File: xml/System.Net/FileWebRequest.xml (hunk 7 above)
Comment (Jozkee) [reply xml/System.Net/FileWebRequest.xml]: Created issue #2672 to address this later.

File: xml/System.Net/FileWebRequest.xml (hunk 3 above)
Comment (carlossanlop) [reply xml/System.Net/FileWebRequest.xml]: @scalablecory @Jozkee there is a neat feature to help provide code suggestions that can be bulk merged:

```suggestion
        <returns>The <see cref="T:System.Threading.Tasks.Task" /> object representing the asynchronous operation.</returns>
```

Comment (carlossanlop) [reply xml/System.Net/FileWebRequest.xml]: Note: The comment is already outdated so you may ignore it.

File: xml/System.Net/IPEndPoint.xml (hunk 5 above)
Comment (carlossanlop) [reply xml/System.Net/IPEndPoint.xml]: Remarks are optional, @scalablecory

File: xml/System.Net/IPEndPoint.xml [hunk 12]
```diff
@@ -764,6 +806,44 @@
         </remarks>
       </Docs>
     </Member>
+    <MemberGroup MemberName="TryParse">
+      <AssemblyInfo>
```
Comment (carlossanlop) [inline xml/System.Net/IPEndPoint.xml]: Why did you add `AssemblyInfo` items manually? These items are supposed to be auto-generated by the Docs repo.

File: xml/System.Net/IPEndPoint.xml [hunk 13]
```diff
@@ -602,10 +642,12 @@
         <Parameter Name="s" Type="System.String" Index="0" FrameworkAlternate="netcore-3.0" />
       </Parameters>
       <Docs>
-        <param name="s">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="s">A <see cref="T:System.String" /> that contains an IP endpoint in dotted-quad notation or network byte order for IPv4 and in colon-hexadecimal notation for IPv6.</param>
+        <summary>Converts an IP network endpoint (address and port) represented as a <see cref="T:System.String" /> to an <see cref="T:System.Net.IPEndPoint" /> instance.</summary>
+        <returns>The object representation of an IP network endpoint</returns>
```
Comment (carlossanlop) [inline xml/System.Net/IPEndPoint.xml]:
```suggestion
        <returns>The object representation of an IP network endpoint.</returns>
```

File: xml/System.Net/IPEndPoint.xml (hunk 12 above)
Comment (carlossanlop) [reply xml/System.Net/IPEndPoint.xml]: Same question about the other `AssemblyInfo` items in other sections of the PR.

Comment (Jozkee) [reply xml/System.Net/IPEndPoint.xml]: Because this is a `MemberGroup` which is not auto-generated. I also did it based on suggested example in https://github.com/dotnet/dotnet-api-docs/pull/2663#discussion_r298781159, which adds the `MemberGroup` node with an `AssemblyInfo` inside. 

But I think you are right, I removed the `AssemblyInfo` nodes and I was able to compile locally without problems, however I would like to have validation from @mairaw or @rpetrusha about this.

Comment (mairaw) [reply xml/System.Net/IPEndPoint.xml]: Pinging @joelmartinez on this one. I also added this manually to other member groups just because others I saw had it.

Joel, is it okay to remove `AssemblyInfo` from the MemberGroup elements?

Comment (joelmartinez) [reply xml/System.Net/IPEndPoint.xml]: @mairaw, yes, that is fine ... the membergroup, from a metadata perspective, is only informative; so it doesn't really conform to any of the other requirements reserved for members :)

**Comment by Jozkee** [comment]
This is ready for re-review:
* Added missing dot.
* Removed `AssemblyInfo` from `MemberGroup`s.
* As per https://github.com/dotnet/dotnet-api-docs/pull/2767#discussion_r304644461, first exception conditions should be written in the same line than `exception` tag.

@mairaw @rpetrusha

**Comment by Jozkee** [comment]
@mairaw, @rpetrusha, Its been a while since I've addressed the comments on this one, can we merge it?

**Comment by mairaw** [comment]
Taking a look now @Jozkee!

**Comment by mairaw** [comment]
@Jozkee I've made some changes to the three files. Can you make sure I didn't introduce anything wrong? Also let me know if you dislike any of the changes I made. Thanks!

File: xml/System.Net/CookieCollection.xml [hunk 14]
```diff
@@ -860,9 +879,20 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Gets an enumerator that can iterate through the <see cref="T:System.Net.CookieCollection" />.</summary>
+        <returns>An <see cref="T:System.Collections.Generic.IEnumerator`1" /> for this collection.</returns>
```
Comment (Jozkee) [inline xml/System.Net/CookieCollection.xml]: Same as GetEnumerator in line 527.
```suggestion
        <returns>A generic enumerator for this collection.</returns>
```

Comment (mairaw) [reply xml/System.Net/CookieCollection.xml]: I think we can keep this one as-is

File: xml/System.Net/CookieCollection.xml [hunk 15]
```diff
@@ -515,7 +524,7 @@
       <Parameters />
       <Docs>
         <summary>Gets an enumerator that can iterate through a <see cref="T:System.Net.CookieCollection" />.</summary>
-        <returns>An instance of an implementation of an <see cref="T:System.Collections.IEnumerator" /> interface that can iterate through a <see cref="T:System.Net.CookieCollection" />.</returns>
+        <returns>An <see cref="T:System.Collections.Generic.IEnumerator" /> for this collection.</returns>
```
[the remark this answers is not in the document]
Comment (mairaw) [reply xml/System.Net/CookieCollection.xml]: I've added to match what we had for similar APIs. I just simplified the sentence a bit, did not add the cref.

**Comment by Jozkee** [comment]
@mairaw Looks good to me, can we merge?

**Comment by carlossanlop** [comment]
@mairaw thanks for your last commit to fix the broken cref. The latest build passed without warnings. Is this good to merge?

**Comment by carlossanlop** [comment]
@mairaw successful build with no warnings. I think it's ready to merge.
